### PR TITLE
added separate method to encrypt string and binary

### PR DESCRIPTION
**Pull request by vijayshiyani**

### CODE REVIEW

File: src/decryption/decryption.ts [hunk 1]
```diff
@@ -10,13 +10,55 @@ interface IEncryptionOptions {
...
+ */
+export async function decryptWithKey(
+  {
+    serialized,
+    key,
+  }: {
+    serialized: string;
+    key: string;
+  },
+  encoding?: Encoding
```
Comment (zbarbuto) [inline src/decryption/decryption.ts]: Should we provide a default encoding here of `raw` so the method still functions as it did before when upgrading?

File: src/decryption/decryption.ts [hunk 2]
```diff
@@ -79,11 +134,38 @@ function _deriveKeyWithOptions(
...
+ * The default encoding is 'raw' is suitable for binaries and 1 byte UTF-8.
+ * string with greater than 2 bytes UTF-8 string will produce an incorrect result. e.g. data string '鍵键'
+ * encrypted with 'raw' encoding will produce incorrect decrypted value.
+ */
+export function decryptWithKeyUsingArtefacts(
+  key: string,
+  encryptedData: any,
+  strategy: CipherStrategy,
+  { iv, at, ad }: IEncryptionOptions,
+  encoding?: Encoding
```
Comment (zbarbuto) [inline src/decryption/decryption.ts]: Here too

File: src/encryption/encryption.ts [hunk 3]
```diff
@@ -41,42 +41,129 @@ export interface IEncryptionResult {
...
+ * @deprecated This method should be replaced by
+ * encryptBinaryWithGeneratedKey or encryptStringWithGeneratedKey method.
+ * The default encoding is 'raw' is suitable for binaries and 1 byte UTF-8.
+ * string with greater than 2 bytes UTF-8 string will produce an incorrect result. e.g. data string '鍵键'
+ * encrypted with 'raw' encoding will produce incorrect decrypted value.
+ */
+export async function encryptWithGeneratedKey(
+  options: IEncryptionOptionsWithoutKey,
+  serializationVersion: SerializationFormat = SerializationFormat.latest_version,
+  encoding?: Encoding
```
Comment (zbarbuto) [inline src/encryption/encryption.ts]: ||

File: src/encryption/encryption.ts [hunk 4]
```diff
@@ -41,42 +41,129 @@ export interface IEncryptionResult {
...
+ * @deprecated This method should be replaced by
+ * encryptStringWithKeyDerivedFromString or encryptBinaryWithKeyDerivedFromString method.
+ * The default encoding is 'raw' is suitable for binaries and 1 byte UTF-8.
+ * string with greater than 2 bytes UTF-8 string will produce an incorrect result. e.g. data string '鍵键'
+ * encrypted with 'raw' encoding will produce incorrect decrypted value.
+ */
+export async function encryptWithKeyDerivedFromString(
+  options: IEncryptionOptions,
+  serializationVersion: SerializationFormat = SerializationFormat.latest_version,
+  encoding?: Encoding
```
Comment (zbarbuto) [inline src/encryption/encryption.ts]: ||

File: src/encryption/encryption.ts [hunk 5]
```diff
@@ -96,14 +183,48 @@ export async function encryptWithKeyDerivedFromString(
...
+ * @deprecated This method should be replaced by
+ * encryptStringWithKey or encryptBinaryWithKey method.
+ * The default encoding is 'raw' is suitable for binaries and 1 byte UTF-8.
+ * string with greater than 2 bytes UTF-8 string will produce an incorrect result. e.g. data string '鍵键'
+ * encrypted with 'raw' encoding will produce incorrect decrypted value.
+ */
+export async function encryptWithKey(
+  { key, data, strategy, iv }: IEncryptionOptions,
+  serializationVersion: SerializationFormat = SerializationFormat.latest_version,
+  encoding?: Encoding
```
Comment (zbarbuto) [inline src/encryption/encryption.ts]: ||

File: src/encryption/encryption.ts [hunk 6]
```diff
@@ -124,22 +245,54 @@ export async function encryptWithKey(
...
+ * The default encoding is 'raw' is suitable for binaries and 1 byte UTF-8.
+ * string with greater than 2 bytes UTF-8 string will produce an incorrect result. e.g. data string '鍵键'
+ * encrypted with 'raw' encoding will produce incorrect decrypted value.
+ */
+export function encryptWithKeyUsingArtefacts(
+  key: string,
+  data: string,
+  strategy: CipherStrategy,
+  iv?: string,
+  encoding?: Encoding
```
Comment (zbarbuto) [inline src/encryption/encryption.ts]: ||

File: src/decryption/decryption.ts (hunk 1 above)
Comment (vijayshiyani) [reply src/decryption/decryption.ts]: added default to 'raw' in all missing places

**Review by zbarbuto — APPROVED** [review]

**Pull request merged**